x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/saml/RestSamlCompleteLogoutAction.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.rest.action.saml;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.BytesRestResponse;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.RestResponse;
+import org.elasticsearch.rest.RestStatus;
+import org.elasticsearch.rest.action.RestBuilderListener;
+import org.elasticsearch.xpack.core.security.action.saml.SamlCompleteLogoutRequestBuilder;
+import org.elasticsearch.xpack.core.security.action.saml.SamlCompleteLogoutResponse;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+
+/**
+ * This Rest endpoint handles SAML LogoutResponse sent from idP with either HTTP-Redirect or HTTP-Post binding.
+ * For HTTP-Redirect binding, it expects {@link Input#queryString} be set to the query string of the redirect URI.
+ * For HTTP-Post binding, it expects {@link Input#content} be set to the value of SAMLResponse form parameter, i.e.
+ * caller of this API must do the work to extract the SAMLResponse value from body of the HTTP-Post request. The
+ * value must also be URL decoded if necessary.
+ */
+public class RestSamlCompleteLogoutAction extends SamlBaseRestHandler{
+
+    private static final Logger logger = LogManager.getLogger(RestSamlCompleteLogoutAction.class);
+
+    static class Input {
```

Comment:
Why do we have another class here, rather than just using `SamlCompleteLogoutRequest` ?

Reply:
Because of copy/paste ... removed it. thanks